bitcoin/psbt.c

```diff
@@ -544,6 +545,119 @@ struct amount_sat psbt_output_get_amount(struct wally_psbt *psbt,
 	assert(amount_asset_is_main(&asset));
 	return amount_asset_to_sat(&asset);
 }
+
+static void add(u8 **key, const void *mem, size_t len)
+{
+	size_t oldlen = tal_count(*key);
+	tal_resize(key, oldlen + len);
+	memcpy(*key + oldlen, mem, len);
```

Comment:
Let's add a memcheck() around this, so if there's something added which is uninitialized, valgrind backtraces to the caller (not later when we use it):

`memcpy(*key + oldlen, memcheck(mem, len), len)`